File: P36 Valid Sudoku/Solution.java
class Solution {
    public boolean isValidSudoku(char[][] board) {
        for(int i = 0; i < 9; i++) {
            HashSet<Character> row = new HashSet<>();
            HashSet<Character> column = new HashSet<>();
            HashSet<Character> box = new HashSet<>();
            for(int j = 0; j < 9; j++) {
                if(board[i][j] != '.' && !row.add(board[i][j])) {
                    return false;
                }
                if(board[j][i] != '.' && !column.add(board[j][i])) {
                    return false;
                }
                int boxRowIndex = 3*(i/3) + j/3;
                int boxColumnIndex = 3*(i%3) + j%3;
                if(board[boxRowIndex][boxColumnIndex] != '.' && !box.add(board[boxRowIndex][boxColumnIndex])) {
                    return false;
                }
            }
        }
        return true;
    }
}

File: P36 Valid Sudoku/solution.py
class Solution:
    def isValidSudoku(self, board: List[List[str]]) -> bool:
        for i in range(9):
            row = set()
            column = set()
            box = set()
            for j in range(9):
                if board[i][j] != '.':
                    if board[i][j] not in row:
                        row.add(board[i][j])
                    else:
                        return False
                if board[j][i] != '.':
                    if board[j][i] not in column:
                        column.add(board[j][i])
                    else:
                        return False
                box_row_index = 3 * (i // 3) + j // 3
                box_column_index = 3 * (i % 3) + j % 3
                if board[box_row_index][box_column_index] != '.':
                    if board[box_row_index][box_column_index] not in box:
                        box.add(board[box_row_index][box_column_index])
                    else:
                        return False
        return True  